src/aws/instancemanager/instancemanager.py

```diff
@@ -281,6 +295,25 @@ async def run(self):
 
                 self.check_all_living()
 
+                # Check if some worker is underloaded or overloaded.
+                underloaded = self.timewindow.get_underloaded()
+                overloaded = self.timewindow.get_overloaded()
+                print("Overloaded: {}, Underloaded: {}".format(overloaded, underloaded))
+                if underloaded:
+                    if overloaded:
+                        pass  # The workload should be balanced. IM should do nothing.
+                    else:
+                        # Kill an underloaded worker. There is not enough work.
+                        self._kill_instance(underloaded[0], instance_types=['worker'])
+                else:
+                    if overloaded:
+                        # We need a new worker. There is too much work.
+                        worker_id = self.start_worker()
+                        self.timewindow.add_empty(worker_id)
```

Comment:
Will this new worker be started as well or does it do that automatically? (or is it already in here somewhere?)

Reply:
`self.start_worker()` does the trick. The start signal to the worker itself comes as soon as the worker started.